### PR TITLE
adding new timer restart functionality

### PR DESCRIPTION
**Pull request by AhmedAbouelkher**
I have added on restart callback to the controller and the main timer class

### CODE REVIEW

**Comment by WiRight** [comment]
Thank you for your Pull Request!
I'll rewiew this soon

File: lib/timer_controller.dart [hunk 1]
```diff
@@ -40,4 +46,17 @@ class CountdownController {
...
+  ///
+  /// restart timer from cold
+  ///
+  restart() {
+    if (this.onRestart != null) {
+      this.onRestart();
+    }
+  }
+
+  serOnRestart(VoidCallback onRestart) {
```
Comment (WiRight) [inline lib/timer_controller.dart]: Maybe it should be `setOnRestart` ?

File: lib/timer_count_down.dart [hunk 2]
```diff
@@ -1,9 +1,8 @@
 import 'dart:async';
 
+import 'package:elhariry/Widgets/Timer/timer_controller.dart';
```
Comment (WiRight) [inline lib/timer_count_down.dart]: Please, make sure to use `import 'package:timer_count_down/timer_controller.dart';`

File: lib/timer_controller.dart [hunk 3]
```diff
@@ -10,6 +10,12 @@ class CountdownController {
   // Called when called `resume` method
   VoidCallback onResume;
 
+  // called to restart the timer
+  VoidCallback onRestart;
+
+  //is timer active
+  bool isCompleted = false;
```
Comment (WiRight) [inline lib/timer_controller.dart]: This param is not used
Maybe it should be removed

**Review by WiRight — CHANGES_REQUESTED** [review]
Please, see comments

File: lib/timer_controller.dart (hunk 1 above)
Comment (AhmedAbouelkher) [reply lib/timer_controller.dart]: yes, it is a typo

File: lib/timer_controller.dart (hunk 3 above)
Comment (AhmedAbouelkher) [reply lib/timer_controller.dart]: I wanted to implement the "isCompleted" to be able to toggle Restart and Resume.
for example:
`_controller.isCompleted ? _controller.restert : _contoller.resume`

File: lib/timer_count_down.dart (hunk 2 above)
Comment (AhmedAbouelkher) [reply lib/timer_count_down.dart]: because I used the package outside its main Flutter packaging folder.

Comment (WiRight) [reply lib/timer_count_down.dart]: okay, before merging, please, remove this

File: lib/timer_controller.dart (hunk 3 above)
Comment (WiRight) [reply lib/timer_controller.dart]: If you want and can, may you add all functionality? 
That was be really cool!

File: lib/timer_count_down.dart (hunk 2 above)
Comment (AhmedAbouelkher) [reply lib/timer_count_down.dart]: I have, you can check it out

File: lib/timer_controller.dart (hunk 3 above)
Comment (AhmedAbouelkher) [reply lib/timer_controller.dart]: I have added it today, you can check it out.

Comment (WiRight) [reply lib/timer_controller.dart]: okay, i'll check it!

**Review by WiRight — APPROVED** [review]

**Review by WiRight — APPROVED** [review]
i think it's ready to merge!

**Comment by WiRight** [comment]
Thank you, for your contribution!

**Pull request merged**